scripts/wsrep_sst_clone.sh

```diff
@@ -0,0 +1,1113 @@
+    check_client_version $CLIENT_VERSION
+
+    readonly ADMIN_USER=$WSREP_SST_OPT_USER
+    readonly ADMIN_PSWD=$WSREP_SST_OPT_PSWD
+
+    MYSQL_RCLIENT="$MYSQL_CLIENT -u$WSREP_SST_OPT_REMOTE_JOINER_USER -h$SST_HOST_STRIPPED \
+                   -P$WSREP_SST_OPT_REMOTE_HOSTPORT --batch --skip_column_names --silent"
+    MYSQL_ACLIENT="$MYSQL_CLIENT -u$ADMIN_USER -S$WSREP_SST_OPT_SOCKET \
+                   --batch --skip_column_names --silent"
+
+    export MYSQL_PWD=$ADMIN_PSWD
+    readonly WSREP_SST_OPT_LPORT=`$MYSQL_ACLIENT -e "select @@port"`
+
+    wsrep_log_debug "-> WSREP_SST_OPT_LPORT: $WSREP_SST_OPT_LPORT "
+    wsrep_log_debug "-> MYSQL_ACLIENT: $MYSQL_ACLIENT "
+    wsrep_log_debug "-> MYSQL_ACLIENT ADMIN_PSWD: $ADMIN_PSWD"
+    wsrep_log_debug "-> MYSQL_RCLIENT: $MYSQL_RCLIENT "
+    wsrep_log_debug "-> MYSQL_RUSER: $WSREP_SST_OPT_REMOTE_JOINER_USER | PWD  $WSREP_SST_OPT_REMOTE_JOINER_PSWD"
+
+
+    if [ $WSREP_SST_OPT_BYPASS -eq 0 ]
+    then
+        #
+        #  Prepare DONOR for cloning
+        #
+
+        # Use script file to avoid sensitive information on the command line
+        CLONE_EXECUTE_SQL=$(wsrep_mktemp_in_dir "$WSREP_SST_OPT_DATA" --suffix=.sql clone_execute_XXXX)
+        CLONE_PREPARE_SQL=$(wsrep_mktemp_in_dir "$WSREP_SST_OPT_DATA" --suffix=.sql clone_prepare_XXXX)
+
+        cleanup_donor # Remove potentially existing clone user
+
+        export MYSQL_PWD=$ADMIN_PSWD
+        wsrep_log_debug "-> PREPARE DONOR"
+        setup_clone_plugin "donor"
+
+#        wsrep_log_info "REQUIRE_SSL=$REQUIRE_SSL, CLIENT_SSL_OPTIONS=$CLIENT_SSL_OPTIONS"
+        wsrep_log_debug "-> PREPARED DONE"
+
+cat << EOF > "$CLONE_PREPARE_SQL"
+CREATE USER "$WSREP_SST_OPT_REMOTE_JOINER_USER"@"%" IDENTIFIED BY '$WSREP_SST_OPT_REMOTE_JOINER_PSWD' $REQUIRE_SSL;
+GRANT BACKUP_ADMIN ON *.* TO "$WSREP_SST_OPT_REMOTE_JOINER_USER"@"%";
+GRANT SELECT ON performance_schema.* TO "$WSREP_SST_OPT_REMOTE_JOINER_USER"@"%";
+GRANT EXECUTE ON *.* TO "$WSREP_SST_OPT_REMOTE_JOINER_USER"@"%";
+EOF
+        RC=0
+        export MYSQL_PWD=$ADMIN_PSWD
+        wsrep_log_debug "-> connecting to donor: $CLONE_PREPARE_SQL $MYSQL_PWD $MYSQL_ACLIENT"
+        $MYSQL_ACLIENT --connect-timeout=60 < $CLONE_PREPARE_SQL || RC=$?
+        wsrep_log_debug "-> $RC connect to donor done"
+
+        if [ "$RC" -ne 0 ]
+        then
+            wsrep_log_error "Donor prepare returned code $RC"
+            cat $CLONE_PREPARE_SQL >> /dev/stderr
+            exit $RC
+        fi
+        #Before waiting for the Joiner Clone mysql we send out the message this is a SST
+        wsrep_log_debug "-> NETCAT signal to nc -w 1 $SST_HOST_STRIPPED $TSST_PORT"
+        echo "SST@$WSREP_SST_OPT_GTID" | nc -w 1 $SST_HOST_STRIPPED $TSST_PORT || :
+
+        # we stay on hold now, waiting for the Joiner to expose the service
+        wsrep_log_info "-> WAIT for Joiner MySQL to be available nc -w 1 -i 1 $SST_HOST_STRIPPED $WSREP_SST_OPT_REMOTE_HOSTPORT"
+        while [ 1 == 1 ]; do
```

Comment:
For better readability
```suggestion
        while true; do
```